fontforge/splineorder2.c

```diff
@@ -1049,9 +1049,12 @@ void SCConvertLayerToOrder2(SplineChar *sc,int layer) {
     if ( sc==NULL )
 return;
 
-    new = SplineSetsTTFApprox(sc->layers[layer].splines);
-    SplinePointListsFree(sc->layers[layer].splines);
-    sc->layers[layer].splines = new;
+    if ( sc->layers[layer].splines != NULL && 
+         sc->layers[layer].splines->spiro_cnt <= 0 ) {
+        new = SplineSetsTTFApprox(sc->layers[layer].splines);
```

Comment:
I guess this isn't a very fair question, but have you dug around enough to identify why this works? It seems like this change amounts to "if there are Spiro points, don't convert". So when copying to a quadratic layer not in Spiro mode, what code winds up doing the "translation"?

Reply:
(I'm asking because ideally a reviewer should understand specifically why the change helps, so it would be faster for me to review if I had some pointers to the other relevant code.)

Reply:
It works because the translation isn't actually necessary for a Spiro spline; the spline is only necessary for *display to the user*, so it doesn't matter if FontForge internally keeps a cubic spline on a quadratic layer. Indeed, unless #3679 is merged, Spiro splines are saved as cubic, which is another bug entirely.

Reply:
When it's actually time to export the font, FontForge will throw away the Spiros and do the proper conversion anyway. So Spiro is a special case; and a bit of a buggy one.

Reply:
But my question relates to the case we've been discussing in the general comment thread: A user has two fonts open, and a CharView open for a glyph in each font. One window has a Cubic foreground but is being edited in Spiro mode. A user copies content in that window. The other window has a Quadratic foreground and is **not** being edited in Spiro mode. The user pastes into that window. For that case to work correctly, if the latter window remains in "Bezier mode", *something* needs to convert the spline to quadratic format. The code in this push request just prevents conversion in certain cases. So the case in question works, something else must do the conversion to quadratic/order2. I'm wondering what that is.

Reply:
> something needs to convert the spline to quadratic format

No it doesn't. It works fine, try it yourself. It looks a little funny at first, yes, but as soon as you edit the Spiro spline it becomes quadratic.

Reply:
I just tried it; it works. I wasn't doubting that, I was asking how it worked. This isn't really code I've looked at closely so I'll leave the review to others.

Reply:
@skef Because `CVMouseDownPoint` calls `SplineSetSpirosClear`. FontForge is good about clearing the Spiros on edits, I'm sure other places also call it. Just look for calls of `SplineSetSpirosClear`

Reply:
After clearing the Spiros, `CVMouseDownPoint` calls `SplineMake`, which takes `order2` which comes from the layer info.

Reply:
@skef Sorry if I'm just not understanding you, sorry you no longer want to review this :(

Reply:
(If no one else has yet) I'll take a look at these pointers later and see if I can situate myself in the code enough to confidently review it.